File: Java/Address/Student.java
//********************************************************************
//  Student.java       Author: Lewis/Loftus
//
//  Represents a college student.
//********************************************************************

//Warren Seto
//Period 2

public class Student
{
   private String firstName, lastName;
   private Address homeAddress, schoolAddress;
   private int grade;
   private int grade1;
   private int grade2;
   private int grade3;
   private double avg;

   //-----------------------------------------------------------------
   //  Sets up this Student object with the specified initial values.
   //-----------------------------------------------------------------
   public Student (String first, String last, Address home, Address school)
   {
      firstName = first;
      lastName = last;
      homeAddress = home;
      schoolAddress = school;
      grade1 = 0;
      grade2 = 0;
      grade3 = 0;
      avg = 0;
   }
   //-----------------------------------------------------------------
   //  Sets the test score for each score
   //----------------------------------------------------------------- 
   public void setTestScore(int num, int grade) 
   {   
	   if (num == 1)
		   grade1 = grade;
	   
	   if (num == 2)
		   grade2 = grade;
	   
	   if (num == 3)
		   grade3 = grade;
   }
   //-----------------------------------------------------------------
   //  Calculates the average
   //----------------------------------------------------------------- 
   public double average ()
   {
	  return avg = (grade1 + grade2 + grade3) / 3;
   }  
   //-----------------------------------------------------------------
   //  Returns a Student's specific grade.
   //----------------------------------------------------------------- 
   public double getTestScore (int choose) 
   {
	   if (choose == 1)
		   grade = grade1;
	   
	   if (choose == 2)
		   grade = grade2;
	   
	   if (choose == 3)
		   grade = grade3;
	   
	   return grade;
   }
   //-----------------------------------------------------------------
   //  Returns this Student object as a string.
   //-----------------------------------------------------------------
   public String toString()
   {
      String result;

      result = firstName + " " + lastName + "\n";
      result += "Home Address: " + homeAddress + "\n";
      result += "School Address: " + schoolAddress + "\n";
      result += "Test Scores: " + grade1 + ", " + grade2 + ", " + grade3 + "\n";
      result += "Average of test scores: " + avg;

      return result;
   }
}

File: ObjectiveC/Address/Address/Student.m
//
//  Student.m
//  Address
//
//  Copyright © 2017 Warren Seto. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "Student.h"

@implementation Student

/* @brief : Sets up this Student object with the specified initial values */
-(instancetype) initWithStudentFirstName:(NSString*)first LastName:(NSString*)last homeAddress:(Address*)home andSchoolAddress:(Address*)school
{
    self = [super init];
    
    _firstName = first;
    _lastName = last;
    _homeAddress = home;
    _schoolAddress = school;
    
    _grade1 = 0;
    _grade2 = 0;
    _grade3 = 0;
    _avg = 0;
    
    return self;
}

/* @brief : Sets the test score for each score */
-(void) setTestScoreWithGradeNumber:(int)num andGrade:(int)grade
{
    if (num == 1)
        _grade1 = grade;
	   
    if (num == 2)
        _grade2 = grade;
	   
    if (num == 3)
        _grade3 = grade;
}

/* @brief : Calculates the average */
-(double) getAverage
{
    _avg = (_grade1 + _grade2 + _grade3) / 3.0;
    return _avg;
}

/* @brief : Returns a Student's specific grade from 1 to 3. If not in range, returns 0 */
-(int) getTestScoreWithScoreNumber:(int)choose
{
    if (choose == 1)
        return _grade1;
	   
    if (choose == 2)
        return _grade2;
	   
    if (choose == 3)
        return _grade3;
    
    return -1;
}

/* @brief : Returns this Student object as a string. */
-(NSString*) description
{
    NSString* result = [NSString stringWithFormat: @"%@ %@\nHome Address: %@\nSchool Address: %@\nTest Scores: %d, %d, %d\nAverage of Test Scores: %f",
                                                        _firstName, _lastName,
                                                        [_homeAddress description],
                                                        [_schoolAddress description],
                                                        _grade1, _grade2, _grade3, [self getAverage]];
    
    return result;
}

@end
